consensus/src/ordered_broadcast/engine.rs

```diff
@@ -547,6 +565,8 @@ impl<
         threshold: group::Signature,
     ) {
         // Set the threshold signature, returning early if it already exists
+        //
+        // TODO: this prevents us from ever sending a finalization message recovered from a partial signature
```

Comment:
We need to refactor the handling of `handle_threshold` to allow messages to be sent when a partial signature is recovered.